config/routes.rb

```diff
@@ -15,9 +15,12 @@
     resources :lessons, path: "", only: [:show]
   end
 
-  resources :lesson_plans, path: "/lesson-plans", only: [:create, :update, :show] do
+  resources :lesson_plans, path: "/lesson-plans", only: [:create, :update] do
     post :lessons, to: "lesson_plans#create_lesson"
     delete :lessons, to: "lesson_plans#destroy_lesson"
+    collection do
+      get ":key", to: "lesson_plans#show"
```

Comment:
Aah I was hoping you could use `as: "share!"` to get the right helper name, but that raises an error. I didn't think to change the segment to `:key` -- that's a nice detail.